### PR TITLE
[vcpkg] Increment number of retries and timeout of fs.rename()

### PR DESCRIPTION
**Pull request by mapozyan**
fs.rename() might fail for many different reasons.

As explained in #3576, one of the reasons might be a virus scanner. But unfortunately, disabling or adding exception is not always an option (think of domain-controlled policies for example, or the archive might indeed contain a virus!).

So if we assume that we need to retry fs.rename() for whatever reason, then we probably need to make retries more useful.

Right now, the code below in master version translates to:

```cpp
fs.rename();
sleep(100ms);
fs.rename();
sleep(300ms);
fs.rename();
```

(needless to say that `i++;` in `for` loop is looking quite confusing)

When the file system is under high load, timeout of some hundreds of milliseconds does not make much sense.

So I think it would be better to change retries to 0s-1s-2s-3s-4s scheme.

### CODE REVIEW

**Comment by BillyONeal** [comment]
Does making the retry longer actually fix the virus scanner problem? If it was blocked it seems unlikely that waiting longer would make the block go away

**Comment by mapozyan** [comment]
Antivirus is blocking the file just while its doing actual archive scanning. Once the scanning is done the file is unlocked.

**Comment by BillyONeal** [comment]
That suggests that there is no timeout value that can be chosen that would make this truly reliable, but we could do better.

If we're increasing these timeouts, we should probably use exponential backoff rather than linear backoff; then setting high enough timeouts to give you more likely reliability might help.

**Comment by mapozyan** [comment]
Agree, makes sense. What about 500ms-1s-2s-4s-8s timeouts?

**Comment by BillyONeal** [comment]
> Agree, makes sense. What about 500ms-1s-2s-4s-8s timeouts?

How about 8ms, 16ms, 32ms, 64ms, .... etc.? Injecting a half second sleep into places that might have succeeded previously much faster seems like a regression.

**Comment by strega-nil** [comment]
(* poke @mapozyan *)

File: toolsrc/src/vcpkg/archives.cpp [hunk 1]
```diff
@@ -97,12 +98,17 @@ namespace vcpkg::Archives
 
         fs.rename(to_path_partial, to_path, ec);
 
-        for (int i = 0; i < 5 && ec; i++)
+        using namespace std::chrono_literals;
+
+        auto retry_delay = 8ms;
+
+        for (int i = 0; i < 10 && ec; i++)
```
Comment (JackBoosY) [inline toolsrc/src/vcpkg/archives.cpp:105]:
```suggestion
        for (int i = 0; i < 5 && ec; i++)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
I think we can just retry 5 times.

File: toolsrc/src/vcpkg/archives.cpp (hunk 1 above)
Comment (mapozyan) [reply toolsrc/src/vcpkg/archives.cpp:105]: 5 times means 256ms retry time in total, I am afraid its not enough for AV-scanning of large archive files like cmake.

Comment (JackBoosY) [reply toolsrc/src/vcpkg/archives.cpp:105]: Hmm... fine.

**Review by JackBoosY — APPROVED** [review]

File: toolsrc/src/vcpkg/archives.cpp [hunk 2]
```diff
@@ -97,12 +98,17 @@ namespace vcpkg::Archives
...
         {
-            i++;
             using namespace std::chrono_literals;
-            std::this_thread::sleep_for(i * 100ms);
+            std::this_thread::sleep_for(retry_delay);
             fs.rename(to_path_partial, to_path, ec);
+            System::printf("fs.rename(%s, %s)\n", fs::u8string(to_path_partial), fs::u8string(to_path));
```
Comment (strega-nil) [inline toolsrc/src/vcpkg/archives.cpp:110]:
```suggestion
```

**Comment by strega-nil** [comment]
Awesome, thanks @mapozyan :)

**Pull request merged**